File: API/ios-tms.API/ViewModels/Quotes/QuoteRateViewModel.cs
using iOS_TMS.Core.Domain;

namespace iOS_TMS.API.ViewModels.Quotes;

public class QuoteRateViewModel
{
    public int Id { get; set; }
    public int QuoteId { get; set; }
    public string Carrier { get; set; }
    public decimal Rate { get; set; }
    public string RateFormatted => Rate.ToString("0.00");
    public decimal Cost { get; set; }
    public string CostFormatted => Cost.ToString("0.00");
    public DateTime PickupDate { get; set; }
    public string PickupDateFormatted => PickupDate.ToString("MM/dd/yyyy");
    public DateTime EstDeliveryDate { get; set; }
    public string EstDeliveryDateFormatted => EstDeliveryDate.ToString("MM/dd/yyyy");

    public static QuoteRateViewModel From(QuoteRate quoteRate)
    {
        return new QuoteRateViewModel
        {
            Id = quoteRate.Id,
            QuoteId = quoteRate.QuoteId,
            Carrier = quoteRate.Carrier,
            Rate = quoteRate.Rate,
            Cost = quoteRate.Cost,
            PickupDate = quoteRate.PickupDate,
            EstDeliveryDate = quoteRate.EstDeliveryDate
        };
    }
}

File: App/ios-tms/ViewModels/Quotes/QuoteRateViewModel.swift
//
//  QuoteRateViewModel.swift
//  ios-tms
//
//  Created by Kyle Morton on 9/17/22.
//

import Foundation

struct QuoteRateViewModel : Identifiable, Codable {
    
    var id: Int
    var quoteId: Int
    var carrier: String
    var rate: Decimal
    var rateFormatted: String
    var cost: Decimal
    var costFormatted: String
//    var pickupDate: Date
    var pickupDateFormatted: String
//    var estDeliveryDate: Date
    var estDeliveryDateFormatted: String
    
}
